pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -161,8 +162,18 @@ public BaseCombineOperator run() {
       // Streaming query (only support selection only)
       return new StreamingSelectionOnlyCombineOperator(operators, _queryContext, _executorService, _streamObserver);
     }
+    GapfillUtils.GapfillType gapfillType = GapfillUtils.getGapfillType(_queryContext);
```

Comment:
The reason I need handle the gapfill query first because otherwise I need make sure the query is not gapfill query for last two ifs. I prefer to keep them as they are.